rest_framework_json_api/utils.py

```diff
@@ -236,15 +243,11 @@ def get_related_resource_type(relation):
                     relation_model = parent_model_relation.related.related_model
                 else:
                     relation_model = parent_model_relation.related.model
-            elif parent_model_relation_type is ManyToManyDescriptor:
```

Comment:
What was the driving logic for removing these `elif` clauses? I recall reviewing a PR fairly recently that added some of these cases and I would like to make sure we're not regressing here.